### PR TITLE
add closure that guarantees mutex unlock in loop

### PR DESCRIPTION
**Pull request by juanvallejo**
Attempts to address https://github.com/openshift/origin/issues/12930

Based on output from the linked test flake:
```
00:47:51.777 Running test/cmd/observe.sh:22: executing 'oc observe services --once --all-namespaces' expecting success and text 'default kubernetes'...
05:54:44.989 Connection to 172.18.7.33 closed by remote host.
05:54:44.993 Build step 'Execute shell' marked build as failure
```

It is a possibility that the test is hanging for multiple hours based on a deadlock happening. This patch eliminates that possibility.

cc @openshift/cli-review @enj @smarterclayton

### CODE REVIEW

**Comment by enj** [comment]
The diff is significantly easier to see [when ignoring whitespace](https://github.com/openshift/origin/pull/12980/files?w=1).

**Comment by juanvallejo** [comment]
[test]

**Comment by smarterclayton** [comment]
Yeah the continue was broken.  LGTM

File: pkg/cmd/cli/cmd/observe/observe.go [hunk 1]
```diff
@@ -431,6 +431,7 @@ func (o *ObserveOptions) Run() error {
 		go func() {
 			<-time.After(o.exitAfterPeriod)
 			lock.Lock()
+			defer lock.Unlock()
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/observe/observe.go]: This is unnecessary because we os.Exit(0).  It doesn't hurt, so not an issue.

**Comment by smarterclayton** [comment]
[merge]

**Comment by juanvallejo** [comment]
flaked on https://github.com/openshift/origin/issues/12923 re[test]

**Comment by fabianofranz** [comment]
flaked on https://github.com/openshift/origin/issues/12784#issuecomment-277388590 re[merge]

**Pull request merged**